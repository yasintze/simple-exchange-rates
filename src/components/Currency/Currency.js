// @flow
import React from "react";
import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import DeleteIcon from "@material-ui/icons/Delete";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";
import numeral from "numeral";

type Props = {
  header: string,
  description: number,
  meta: string,
  classes: Object
};

const styles = theme => ({
  button: {
    margin: theme.spacing.unit
  },
  card: {
    minWidth: 275
  },
  paper: {
    padding: theme.spacing.unit * 2,
    textAlign: "center",
    color: theme.palette.text.secondary,
    backgroundColor: "transparent",
    maxWidth: 768,
    margin: "auto"
  },
  textBlack: {
    color: "black"
  }
});

const Currency = (props: Props) => {
  const { header, description, meta, classes } = props;
  const metaLabel = `1 USD = ${meta} ${header}`;

  return (
    <Paper className={classes.paper}>
      <Card className={classes.card}>
        <CardContent>
          <Typography
            variant="h6"
            align="left"
            className={classes.textBlack}
            gutterBottom
          >
            {header}
          </Typography>
          <Typography
            variant="body1"
            align="left"
            className={classes.textBlack}
            gutterBottom
          >
            {metaLabel}
          </Typography>
          <Typography
            variant="subtitle1"
            align="left"
            className={classes.textBlack}
            gutterBottom
          >
            {numeral(description).format("0,0.00")}
          </Typography>
        </CardContent>
        <CardActions>
          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
          >
            <span>Delete</span>
            <DeleteIcon className={classes.rightIcon} />
          </Button>
        </CardActions>
      </Card>
    </Paper>
  );
};

export default withStyles(styles)(Currency);
