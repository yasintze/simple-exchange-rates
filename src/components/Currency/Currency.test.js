// @flow
import React from "react";
import { shallow } from "enzyme";

import Currency from "./Currency";

describe("Currency component", () => {
  it("render correctly", () => {
    const wrapper = shallow(<Currency />);
    expect(wrapper).toMatchSnapshot();
  });
});
