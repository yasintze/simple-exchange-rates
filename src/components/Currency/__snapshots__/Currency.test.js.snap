// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency component render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Currency) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "Currency-button-1",
        "card": "Currency-card-2",
        "paper": "Currency-paper-3",
        "textBlack": "Currency-textBlack-4",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "Currency-button-1",
          "card": "Currency-card-2",
          "paper": "Currency-paper-3",
          "textBlack": "Currency-textBlack-4",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
